src/shared/configuration/applicationInsightsConfig.ts

```diff
@@ -109,6 +123,29 @@ export class ApplicationInsightsConfig implements IConfig {
         this.extendedMetrics[ExtendedMetricType.gc] = false;
         this.extendedMetrics[ExtendedMetricType.heap] = false;
         this.extendedMetrics[ExtendedMetricType.loop] = false;
+        this._resource = this._getDefaultResource();
+    }
+
+    private _getDefaultResource(): Resource {
+        let resource = Resource.EMPTY;
+        resource.attributes[SemanticResourceAttributes.SERVICE_NAME] = DEFAULT_ROLE_NAME;
+        if (process.env.WEBSITE_SITE_NAME) {
```

Comment:
I don't think this default behavior for resource attributes is defined anywhere in [mappings](https://github.com/microsoft/common-schema/blob/main/Mappings/AzureMonitor-AI.md).

Reply:
Nope, we had these default values since first day in Node.js SDK it could be different in other languages.